ingestion/post_ingestion_checks.py

```diff
@@ -0,0 +1,262 @@
+import argparse
+import json
+import os
+
+import datahub.emitter.mce_builder as mce_builder
+import datahub.emitter.mcp_builder as mcp_builder
+from datahub.ingestion.graph.client import DataHubGraph
+from datahub.ingestion.graph.config import DatahubClientConfig
+
+from ingestion.config import ENV, INSTANCE, PLATFORM
+from ingestion.ingestion_utils import (
+    get_cadet_metadata_json,
+    parse_database_and_table_names,
+    validate_fqn,
+)
+
+count_by_platform_graph_query = """
+query platformCounts($input: AggregateAcrossEntitiesInput!) {
+    aggregateAcrossEntities(input: $input) {
+        facets {
+            field
+            displayName
+            aggregations {
+                value
+                count
+            }
+        }
+    }
+}
+"""
+
+
+def _get_table_database_mappings(manifest):
+    mappings = {}
+    for node in manifest["nodes"]:
+        if manifest["nodes"][node]["resource_type"] in ["model", "seed"]:
+            fqn = manifest["nodes"][node]["fqn"]
+            if validate_fqn(fqn):
+                database, table_name = parse_database_and_table_names(
+                    manifest["nodes"][node]
+                )
+
+                dataset_urn = mce_builder.make_dataset_urn_with_platform_instance(
+                    name=f"{database}.{table_name}",
+                    platform=PLATFORM,
+                    platform_instance=INSTANCE,
+                    env=ENV,
+                )
+                database_key = mcp_builder.DatabaseKey(
+                    database=database,
+                    platform=PLATFORM,
+                    instance=INSTANCE,
+                    env=ENV,
+                    backcompat_env_as_instance=True,
+                )
+                database_urn = database_key.as_urn()
+                if "dc_display_in_catalogue" in manifest["nodes"][node]["tags"]:
+                    mappings[dataset_urn] = database_urn
+
+    return mappings
+
+
+def _remove_empty_dicts(d):
+    if not isinstance(d, dict):
+        return d
+    cleaned_dict = {}
+    for k, v in d.items():
+        if isinstance(v, dict):
+            nested = _remove_empty_dicts(v)
+            if nested:  # Only add non-empty dictionaries
+                cleaned_dict[k] = nested
+        elif v:  # Only add non-empty values
```

Comment:
I think we could get rid of the elif if we used a dict comprehension, but happy to keep it as-is.

with dict comp it would be something like this:

`def _remove_empty_dicts(d):
    if not isinstance(d, dict):
        return d
    return {
        k: _remove_empty_dicts(v) if isinstance(v, dict) else v
        for k, v in d.items()
        if not isinstance(v, dict) and v
    }`

Reply:
think this would work, but might be less readable :

```
def _remove_empty_dicts(d):
    if not isinstance(d, dict):
        return d
    return {
        k: _remove_empty_dicts(v) if isinstance(v, dict) else v
        for k, v in d.items()
        if (isinstance(v, dict) and _remove_empty_dicts(v)) or (not isinstance(v, dict) and v)
    }
    ```